sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/PeriodicSequence.java

```diff
@@ -207,7 +213,7 @@ public ProcessContinuation processElement(
 
       boolean claimSuccess = true;
 
-      estimator.setWatermark(Instant.ofEpochMilli(restriction.getFrom()));
+      estimator.setWatermark(Instant.ofEpochMilli(nextOutput));
```

Comment:
This change is made in accordance with Python implementation. Currently the watermark will be set back to the initial timestamp (which violates the monotonically increasing) though it will be set back in the while clause.